changelog.d/20211213_135655_aaschaer_client_login.md

```diff
@@ -0,0 +1,5 @@
+### Enhancements
+
+* Add support for client identities to use the Globus CLI
+  by setting GLOBUS_CLI_CLIENT_ID and GLOBUS_CLI_CLIENT_SECRET
+  environment variables (:pr:`NUMBER`)
```

Comment:
```suggestion
* Add support for client credentials for authentication in the Globus CLI
  by setting GLOBUS_CLI_CLIENT_ID and GLOBUS_CLI_CLIENT_SECRET
  environment variables (:pr:`NUMBER`)
** Both variables must be set to enable this behavior
** Tokens generated with client credentials are cached in the current user's home
   directory, but isolated from any user credentials
** With client credentials, `globus login` is invalid, but `globus logout` can be used
   to revoke any cached tokens
```

I think the changelog is a good place to explain usage, since it's where a lot of people will be looking to see the new feature. Above is a suggested first-pass at fleshing out the text a bit more. We can iterate further on this, but I think we want a bit more text to explain.

We'll also want a page on the docs site, probably. But adding it here is still important.